Mathlib/RingTheory/Localization/FractionRing.lean

```diff
@@ -144,9 +144,19 @@ end CommRing
 variable {B : Type*} [CommRing B] [IsDomain B] [Field K] {L : Type*} [Field L] [Algebra A K]
   [IsFractionRing A K] {g : A →+* L}
 
+private lemma nontrivial_of_ringHom_nontrivial
+  {A B : Type*} [CommRing A] [CommRing B] (f : A →+* B) [Nontrivial B] : Nontrivial A := by
+  rcases subsingleton_or_nontrivial A with _ | h
+  · have : (1 : A) = 0 := Subsingleton.elim _ _
+    apply_fun f at this
+    rw [map_one, map_zero] at this
+    exact False.elim (one_ne_zero this)
+  exact h
```

Comment:
This is `RingHom.domain_nontrivial` right?

Reply:
Thanks!

Reply:
Done.